project-squirrel-dragon/PresentationLayer/Modules/CardDetailsModule/CardDetailViewController.swift

```diff
@@ -0,0 +1,184 @@
+//
+
+import UIKit
+
+class CardDetailViewController: UIViewController {
+
+    // MARK: - Properties
+    @IBOutlet private weak var doneButton: UIButton!
+    @IBOutlet private weak var cancelButton: UIButton!
+    @IBOutlet weak var nationalPokedexNumberLabel: UILabel!
+    @IBOutlet private weak var cardImageView: UIImageView!
+    @IBOutlet private weak var nameLabel: UILabel!
+    @IBOutlet private weak var rarityLabel: UILabel!
+    @IBOutlet private weak var typeLabel: UILabel!
+    @IBOutlet private weak var numberLabel: UILabel!
+    @IBOutlet private weak var pricesLabel: UILabel!
+    @IBOutlet private weak var collectedNumberTextField: UITextField!
+    @IBOutlet private weak var addButton: UIButton!
+    @IBOutlet private weak var minusButton: UIButton!
+
+    @IBOutlet private weak var pricesNumberLabel: UILabel!
+    @IBOutlet private weak var cardFront: UIView!
+    @IBOutlet private weak var cardBack: UIView!
+    @IBOutlet private weak var card: UIView!
+
+
+    fileprivate func flipAnimation(inContainer viewContainer: UIView, fromView view1: UIView, toView view2: UIView, animationOption: UIView.AnimationOptions) {
+        //change hidden state half way through animation using delay
+        UIView.transition(with: viewContainer, duration: 0.5, options: [animationOption], animations:{
+            let delay = 0.25
+            DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + delay, execute: { () -> Void in
+                view1.isHidden = true
+                view2.isHidden = false
+            })
+        }, completion: nil)
+    }
+
+    @IBAction func flipCardButtonPressed(_ sender: UIButton) {
+
+        if cardBack.isHidden == true {
+            flipAnimation(inContainer: self.card,
+                          fromView: cardFront,
+                          toView: cardBack,
+                          animationOption: .transitionFlipFromRight)
+
+            } else if cardFront.isHidden == true {
+                flipAnimation(inContainer: self.card,
+                              fromView: cardBack,
+                              toView: cardFront,
+                              animationOption: .transitionFlipFromLeft)
+            }
+        }
+
+    private lazy var viewModel = CardDetailViewModel()
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        card.layer.cornerRadius = StyleKit.Cards.cornerRadius
+        collectedNumberTextField.keyboardType = .numberPad
+        collectedNumberTextField.delegate = self
+        configureCardImageView()
+        configureNumberLabel()
+        configureNationalPokedexNumberLabel()
+        configureNameLabel()
+        configureRarityLabel()
+        configureTypeLabel()
+        configurePricesLabel()
+        configureTextField()
+        navigationController?.setNavigationBarHidden(true, animated: animated)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
```

Comment:
Same with these Lifecycle functions, best to keep them at the top because other developers know to look for them 😄